src/bin/cybergrape/main.rs

```diff
@@ -56,30 +59,75 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
             )),
         ),
         Serial(serial_command) => (
-            serial_command.num_tags,
+            serial_command.num_tags as usize,
             // outfile is common to both commands, though it means slightly different things
             serial_command.outfile,
             // the serial command doesn't have any audio samples and doesn't need gain/range info
             None,
         ),
     };
 
-    let available_ports = SerialPort::available_ports().expect("Failed to get available ports");
-    let selected_port_opt = device_selector::run_device_selector(available_ports)?;
+    // Figure out what serial port our antena box is on
+    let available_ports = SerialPort::available_ports()?;
+    let selected_port_opt = gui::device_selector(available_ports)?;
     let selected_port = match selected_port_opt {
         Some(port) => port,
-        None => return Ok(()),
+        None => {
+            error!("Port selection failed");
+            return Ok(());
+        }
     };
 
     // Try to open the requested port and set its read timeout to infinity
     // (well, about 584,942,417,355 years, which is close enough)
-    let mut port = SerialPort::open(selected_port, 115200).expect("Failed to open port");
+    let mut port = SerialPort::open(selected_port, BAUD_RATE).expect("Failed to open port");
     port.set_read_timeout(std::time::Duration::MAX)
         .expect("Failed to set read timeout");
 
     let hdm = Arc::new(Mutex::new(Hdm::new()));
-    let mut accumulator = UpdateAccumulator::new(hdm.clone());
+    let th_hdm = hdm.clone();
+
+    listen_on_port(port, hdm);
+
+    if let Some((_sound_data, gains, ranges)) = audio_settings {
+        let sphericalizer = Sphericalizer::new(gains.into_iter().zip(ranges).collect());
```

Comment:
To recap for my understanding, this is:
1. If we're already spatialized ("sphericalized") the data, read the data, its gains, and its ranges back in.
2. Otherwise, spin up an `UpdateAccumulator` to aggregate this data.